apps/dashboard/src/components/amount-input.tsx

```diff
@@ -64,16 +65,26 @@ export const AmountInput = ({
               <FormControl>
                 <Input
                   type="number"
-                  className={cn(
-                    'min-w-[20ch] [appearance:textfield] [&::-webkit-inner-spin-button]:appearance-none [&::-webkit-outer-spin-button]:appearance-none',
-                    className
-                  )}
+                  className="min-w-[20ch] [appearance:textfield] [&::-webkit-inner-spin-button]:appearance-none [&::-webkit-outer-spin-button]:appearance-none"
                   placeholder={placeholder}
                   disabled={isReadOnly}
-                  {...field}
+                  value={field.value}
+                  onKeyDown={(e) => {
+                    if (e.key === 'e' || e.key === '-' || e.key === '+' || e.key === '.' || e.key === ',') {
+                      e.preventDefault();
+                    }
```

Comment:
do not allow typing these characters (they are allowed by default in number inputs)